extensions/amp-story/1.0/amp-story.js

```diff
@@ -556,11 +556,53 @@ export class AmpStory extends AMP.BaseElement {
 
   /** @override */
   layoutCallback() {
-    if (!AmpStory.isBrowserSupported(this.win) && !this.platform_.isBot()) {
+    if (!AmpStory.isBrowserSupported(this.win)
+    && !this.platform_.isBot()) {
       this.storeService_.dispatch(Action.TOGGLE_SUPPORTED_BROWSER, false);
       return Promise.resolve();
     }
+    return this.supportBrowserRender();
+    /*const firstPageEl = user().assertElement(
+        this.element.querySelector('amp-story-page'),
+        'Story must have at least one page.');
+
+    const initialPageId = this.getHistoryStatePageId_() || firstPageEl.id;
+
+    if (!this.paginationButtons_) {
+      this.buildPaginationButtons_();
+    }
+
+    this.initializeBookend_();
+
+    const storyLayoutPromise = this.initializePages_()
+        .then(() => this.buildSystemLayer_())
+        .then(() => {
+          this.pages_.forEach(page => {
+            page.setActive(false);
+          });
+        })
+        .then(() => this.switchTo_(initialPageId))
+        .then(() => this.preloadPagesByDistance_())
+        .then(() => {
+          // Preloads and prerenders the share menu if mobile, where the share
+          // button is visible.
+
+          if (!this.storeService_.get(StateProperty.DESKTOP_STATE)) {
+            this.shareMenu_.build();
+          }
+        });
+
+    // Do not block the layout callback on the completion of these promises, as
+    // that prevents descendents from being laid out (and therefore loaded).
+    storyLayoutPromise.then(() => this.whenPagesLoaded_(PAGE_LOAD_TIMEOUT_MS))
+        .then(() => this.markStoryAsLoaded_());
+
+    this.validateConsent_();
+
+    return storyLayoutPromise;*/
+  }
 
+  supportBrowserRender() {
```

Comment:
Let's add a comment explaining what this function does. Also, how about `layoutStory`?